File: Add Code Here/JAVA/Sorting/InsertionSort.java
class InsertionSort {
    void insertionSort(int arr[]) {
        int n = arr.length;
        for (int i = 1; i < n; i++) {
            int key = arr[i];
            int j = i - 1;
            
            // Move elements of arr[0..i-1] that are greater than key
            // to one position ahead of their current position
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
    }

    // Prints the array
    void printArray(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n; ++i) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String args[]) {
        InsertionSort obj = new InsertionSort();
        int arr[] = {64, 34, 25, 12, 9, 11, 90};
        obj.insertionSort(arr);
        System.out.println("Sorted array");
        obj.printArray(arr);
    }
}

// Time Complexity: O(n^2)
// Space Complexity: O(1)

File: Add Code Here/C++/Sorting/insertion_sort.cpp
#include<bits/stdc++.h>
using namespace std;
void PrintArray(vector<int>& arr, int n){
    for(int i = 0; i < n;i++){
        cout<<arr[i]<<endl;
    }
}

void insertion_sort(vector<int>& arr, int n){
   
           for(int i=1; i < n; i++){
             int j=i-1;
                int  temp =arr[i];
                for( ;j>=0;j--){
                    if(arr[j]>temp){
                         arr[j+1] =arr[j];
                    }
                    else{
                        break;
                    }
                }
                arr[j+1]=temp;
           }
}


int main(){
  vector <int> arr ={10,1,7,6,14,9};
  insertion_sort(arr,6);
  PrintArray(arr,6);
}